lib/queue.rb

```diff
@@ -1,31 +1,61 @@
 class Queue
-
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(30)
+    @front = @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # queue is empty
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+      @store[@front] = element
+      
+    elsif @front == @back
+      #queue is full
+      raise ArgumentError, "Queue is full"
+      
+    else      
+      @store[@back] = element
+      @back = (@back + 1) % @store.length
+    end
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      raise ArgumentError, "Queue is empty"
+    end  
+    
+    removed = @store[@front]
+    @front = (@front + 1) % @store.length  
+    
+    if @front == @back
+      @front = @back = -1
+    end 
+    
+    return removed
   end
-
+  
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
-
+  
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front...@back].length
   end
```

Comment:
This only works if `@front` < `@back`.  What about if `@back` has wrapped around the buffer?